rpc/src/v1/helpers/signature.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethkey::{recover, public_to_address, Signature};
+use jsonrpc_core::Result;
+use v1::types::{Bytes, BasicAccount, H256, U64};
+use v1::helpers::errors;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+
+pub fn verify_signature(is_prefixed: bool, message: Bytes, v: U64, r: H256, s: H256, chain_id: Option<u64>) -> Result<BasicAccount> {
+	let hash = if is_prefixed {
+		eth_data_hash(message.0)
+	} else {
+		keccak(message.0)
+	};
+	let v: u64 = v.into();
+	let is_valid_for_current_chain = match (chain_id, v) {
+		(None, v) if v == 0 || v == 1 => true,
+		(Some(chain_id), v) if v > 36 => (v - 35) / 2 == chain_id,
```

Comment:
if there's no `chain_id`, then `v` must either 1/0 to be a valid for the current chain.
otherwise, if there's a `chain_id` and `v` > 36; (36 is the lowest value, where `v` = 0, `chain_id` = 1)
we check if the encoded `chain_id` is valid for the current chain.

(`v` - 35) / 2 == `chain_id`

Reply:
This is related to extract to chain_id and remove the replay protection right?

This code is hard to read, can you add an inline comment and refer `EIP 155` and perhaps add a one-liner function named `remove_replay_protection` or something similar?